readme.txt
    ///////////////////////////////////////////////////////////
    // razorCMS                                              //
    // readme.txt                                            //
    // GPLv3                                                 //
    // smiffy6969                                            //
    // www.razorcms.co.uk                                    //
    // www.mis-limited.com                                   //
    // 03/2008                                               //
    // ----------------------------------------------------- //
    // V0.1  -  03/2008  -  Version 0.1 first release        //
    // ----------------------------------------------------- //
    // V0.2  -  06/2008  -  New features + bug fix resulting //
    //                      in up issue to V0.2BETA          //
    //                                                       //
    //                      Bug fixes for issue to V0.2RC    //
    // ----------------------------------------------------- //
    // V0.2  -  08/2008  -  RC2 Bug fix release              //
    // ----------------------------------------------------- //
    // V0.3  -  11/2008  -  BETA1 Bug fix and new features   //
    // ----------------------------------------------------- //
    // V0.3  -  02/2009  -  RC2 Bug fix, code walk and some  //
    //                      movement of functions.           //
    // ----------------------------------------------------- //
    // V1.0  -  06/2009  -  ALPHA Security, Bug fix and new  //
    //                      functionality                    //
    // ----------------------------------------------------- //
    // V1.0  -  07/2009  -  BETA Bug fix and new             //
    //                      functionality                    //
    // ----------------------------------------------------- //
    // V1.0  -  08/2009  -  BETA2 Bug fix and new            //
    //                      functionality                    //
    // ----------------------------------------------------- //
    // V1.0  -  09/2009  -  RC Bug fix                       //
    // ----------------------------------------------------- //
    // V1.0  -  10/2009  -  Stable - Bug fix                 //
    // ----------------------------------------------------- //
    // V1.1  -  06/2010  -  Stable - Bug fix + Security fix  //
    // ----------------------------------------------------- //
    ///////////////////////////////////////////////////////////
    
    // Introduction ///////////////////////////////////////////
    
    razorCMS is an open source flat file Content Management System,
    this means there is no need to use an additional database, all
    data is stored in file format.
    
    razorCMS comes in one flavour, core. This is the base system which
    comes with only necessary functionality, just enough to be usefull
    while giving you enough tools to create a website. This means those
    wishing to keep the system fast and small may do so.
    
    Those wishing to have a more flexible easy to use system may then
    begin to add the functionality they wish by using the simple blade
    manager, adding blade packs to the system containing anything from
    sef url re-writes (search engine friendly URL's) to WYSIWYG html
    editors.
    
    This kind of approach to CMS solutions gives you a lego brick
    structure, using only the blade packs you wish to create a tailor
    made solution that fits your needs, without being bogged down with
    functionality you do not need. This should help to keep your
    CMS solution sharp and simple.
    
    PLEASE NOTE : Use of this software is purely at your own risk, the
    author of the software offers no guarantees of the performance or
    suitability of this software and cannot be held responsible for the
    outcome of using this software. By installing and using this software
    you are clearly aknowledging the fact that you the user are solely
    responsible for the use of and any circumstances arrising due to the
    use of this software.    
    
    ///////////////////////////////////////////////////////////

    // About razorCMS /////////////////////////////////////////

    razorCMS came about from some brief time spent using nanoCMS, this
    is an open source CMS solution created by Kalyan Chakravarthy. After
    forking the nanoCMS project to create uCMS, adding extra functionality,
    it was decided that more needed to be done.
    
    razorCMS was created from scratch, using a whole new design, but using
    some of the usefull parts of uCMS in a few places. All of the old
    uCMS code that has found its way into razorCMS has been re-written,
    but most of the code in razorCMS is new code. Version 1.0 of the system
    further distances itself from uCMS showing it to be a completely
    diferent system to uCMS and it's father nanoCMS.
    
    Due to this system using very small parts of code from uCMS, all of razorCMS
    is offered under the GPLv3 licence.
    
    ///////////////////////////////////////////////////////////

    // Installing razorCMS /////////////////////////////////////

    1) Download and unpack razorCMS_core_v#_#.zip.

    2) Upload the install.php file and razorCMS_parcel.zip archive to your server
       using your ftp client or via your hosts file browser application.
       Please ensure you do not unzip the parcel file as this is automatically unpacked
       on installation.

    3) Set the permissions of your installation root (this is the folder your
       install files are placed in) to 0777 open permissions to allow write access
       to this folder so the installer can unpack and install razorCMS. This is only temporary
       and needs to be set back to 0755 or not publicly writable after install, failure to do
       so will leave your installation open to vulnerabilities. You will be prompted
       on the admin homepage if this is not carried out.
    
    4) Using your web browser, navigate to your install location, adding /install.php
       on to the end of the url in the address bar (e.g. www.yoursite.com/install.php).

    5) Follow the instructions on screen to install the system, please ensure that
       you make a note of the administration console username and password you choose,
       you will need to use this to gain access to the administration console to manage
       your new site.

    6) IMPORTANT : ENSURE YOU FOLLOW THE INSTRUCTIONS AT THE END OF THE INSTALL, failure to do so
       may make your site unusable and may leave it open to security issues. Ensure you
       follow ALL instructions on securing your install folder as this is not accessable
       by razorCMS due to standard server configuration.

    ///////////////////////////////////////////////////////////

    // AddOn's ////////////////////////////////////////////////

    razorCMS uses a system called the blade manager to manage addons called blade packs.
    Blade packs offer a way to improve, add to or change the way the system looks
    or acts.

    To add a blade pack simply go to the main website, look through the blade packs
    section, download, and use the dedicated installer section in the blade manager.
    
    Any special requirements needed after the blade pack is installed will be notified
    on the installation screen, like setting up, use and special instructions.

    ///////////////////////////////////////////////////////////    

    // Footnote ///////////////////////////////////////////////

    If you require help, need advice on using razorCMS or want to thank me in some
    way, please contact me through www.razorCMS.co.uk using the support forum.

    This software is provided licenced to the GPLv3 Licence, is freely open for
    alteration, adaptation, manipulation or any other kind of action you can
    think of. Please keep all credit where credit is due, leaving all source
    comments in place.

    For a detailed view of the GPLv3 licence, visit www.gnu.org/licenses/gpl-3.0.html

    It is not necessary to link back to the razorCMS website or morgan integrated
    systems website, to do so would give attackers an insight into the system you
    use, so please feel free to remove any copywrite that is visible from the public
    facing side of the site. However all link backs in the admin side must stay intact
    as these will not pose any threat to security.

    This software is supported and developed for free by morgan integrated systems ltd.

    ///////////////////////////////////////////////////////////
    
    
    
    
    
    
    
    
    
    
    
    
    
    
    
    
    
    
    
    
    
    
    
    
    


release_notes.txt
    ///////////////////////////////////////////////////////////
    // razorCMS                                              //
    // release_notes.txt                                     //
    // GPLv3                                                 //
    // smiffy6969                                            //
    // www.razorcms.co.uk                                    //
    // www.mis-limited.com                                   //
    // 03/2008                                               //
    // ----------------------------------------------------- //
    // V0.1  -  03/2008  -  Version 0.1 first release        //
    //                                                       //
    //                      Some parts loosely based on      //
    //                      nanoCMS V0.3 - with thanks to    //
    //                      Kalyan Chakravarthy              //
    // ----------------------------------------------------- //
    // V0.2  -  06/2008  -  New features + bug fix resulting //
    //                      in up issue to V0.2BETA          //
    //                                                       //
    //                      Bug fixes for issue to V0.2RC    //
    // ----------------------------------------------------- //
    // V0.2  -  08/2008  -  RC2 Bug fix release              //
    // ----------------------------------------------------- //
    // V0.3  -  11/2008  -  BETA1 Bug fix and new features   //
    // ----------------------------------------------------- //
    // V0.3  -  11/2008  -  BETA2 Bug fix and new features   //
    // ----------------------------------------------------- //
    // V0.3  -  12/2008  -  RC Bug fix and code walk         //
    // ----------------------------------------------------- //
    // V0.3  -  02/2009  -  RC2 Bug fix, code walk and some  //
    //                      movement of functions.           //
    // ----------------------------------------------------- //
    // V1.0  -  06/2009  -  ALPHA Security, Bug fix and new  //
    //                      functionality                    //
    // ----------------------------------------------------- //
    // V1.0  -  07/2009  -  BETA Bug fix and new             //
    //                      functionality                    //
    // ----------------------------------------------------- //
    // V1.0  -  08/2009  -  BETA2 Bug fix and new            //
    //                      functionality                    //
    // ----------------------------------------------------- //
    // V1.0  -  09/2009  -  RC Bug fix                       //
    // ----------------------------------------------------- //
    // V1.0  -  10/2009  -  Stable - Bug fix                 //
    // ----------------------------------------------------- //
    // V1.1  -  06/2010  -  Stable - Bug fix + Security fix  //
    // ----------------------------------------------------- //
    ///////////////////////////////////////////////////////////

    // razorCMS v1.1 Stable Release notes /////////////////////

    Below is the full list of files changed from razorCMS core
    v1.0 stable, to razorCMS core v1.1 stable.

    ///////////////////////////////////////////////////////////

    // Changes ////////////////////////////////////////////////

    1) Up-issued project to v1.1 stable
    2) Made changes to stop notifices showing titghtening up code
    3) Fixed bug when not using an editor that converts special chars
    4) Added security fix for script embedding in content over XSS

    ///////////////////////////////////////////////////////////

    // New Files //////////////////////////////////////////////

    ///////////////////////////////////////////////////////////

    // Removed files //////////////////////////////////////////

    datastore/razor_data.txt~

    ///////////////////////////////////////////////////////////

    // Files Altered - core v0.1RC ////////////////////////////

    install.php
    index.php
    core/public_config.php
    core/public_func.php
    admin/lib/unzip.lib.php
    admin/theme/default_admin_xhtml.php
    admin/core/admin_func.php
        
    ///////////////////////////////////////////////////////////

    // Detailed changes of files listed below /////////////////
    
    install.php
        Changes to stop notices popping up for things like variables not registered
        and depricated functions

    index.php
        Changes to stop notices popping up for things like variables not registered
        and depricated functions
    
    core/public_config.php
        Up-issued project to v1.0 stable, change
        $RAZOR['razor_current_ver'] = 'core v1.0 Stable';
        to this
        $RAZOR['razor_current_ver'] = 'core v1.1 Stable';

    core/public_func.php
        Changes to stop notices popping up for things like variables not registered
        and depricated functions

        Fixed error on page not found, defaults to 404, added sockets too so people can 
        overide the header and content output by the error with blade pack

    admin/lib/unzip.lib.php
        Changes to stop notices popping up for things like variables not registered
        and depricated functions

    admin/theme/default_admin_xhtml.php
        Changes to stop notices popping up for things like variables not registered
        and depricated functions  

    admin/core/admin_func.php
        Changes to stop notices popping up for things like variables not registered
        and depricated functions  

        Added in fix in two places to stop the auto convert of html special chars
        when adding new content and editing content when not using an editor

        Added in fix for XSS attacks using random signatures for forms to check identity
        and used this in several places
        

core/public_config.php
<?php
    ///////////////////////////////////////////////////////////
    // razorCMS                                              //
    // core/public_config.php                                //
    // GPLv3                                                 //
    // smiffy6969                                            //
    // www.razorcms.co.uk                                    //
    // www.mis-limited.com                                   //
    // 03/2008                                               //
    // ----------------------------------------------------- //
    // V0.1  -  03/2008  -  Version 0.1 first release        //
    //                                                       //
    //                      Some parts loosely based on      //
    //                      nanoCMS V0.3 - with thanks to    //
    //                      Kalyan Chakravarthy              //
    // ----------------------------------------------------- //
    // V0.2  -  06/2008  -  This file contains changes for   //
    //                      up issue to V0.2RC               //
    // ----------------------------------------------------- //
    // V0.2  -  08/2008  -  RC2 Bug fix release              //
    //                      No changes to this file          //
    // ----------------------------------------------------- //
    // V0.3  -  11/2008  -  BETA1 Bug fix and new features   //
    // ----------------------------------------------------- //
    // V0.3  -  12/2008  -  RC Bug fix and code walk         //
    // ----------------------------------------------------- //
    // V0.3  -  02/2009  -  RC2 Bug fix, code walk and some  //
    //                      movement of functions.           //
    // ----------------------------------------------------- //
    // V1.0  -  06/2009  -  ALPHA Security, Bug fix and new  //
    //                      functionality                    //
    // ----------------------------------------------------- //
    // V1.0  -  07/2009  -  BETA Bug fix and new             //
    //                      functionality                    //
    // ----------------------------------------------------- //
    // V1.0  -  08/2009  -  BETA2 Bug fix and new            //
    //                      functionality                    //
    // ----------------------------------------------------- //
    // V1.0  -  09/2009  -  RC Bug fix                       //
    // ----------------------------------------------------- //
    // V1.0  -  10/2009  -  Stable - Bug fix                 //
    // ----------------------------------------------------- //
    // V1.1  -  06/2010  -  Stable - Bug fix + Security fix  //
    // ----------------------------------------------------- //
    ///////////////////////////////////////////////////////////

    ///////////////////////////////////////
    //   Optional System Configuration   //
    // EDIT LINES BELOW AT YOUR OWN RISK //
    ///////////////////////////////////////

    $RAZOR = array();

    // location of logs directory //
    $RAZOR['logs_dir'] = 'datastore/razor_temp_logs/';

    // location of failed login log //
    $RAZOR['failed_logs'] = 'razor_failed_login.txt';

    // location of datastore directory //
    $RAZOR['datastore_dir'] = 'datastore/';

    // location of system data file razor_data.txt //
    $RAZOR['system_file'] = 'datastore/razor_data.txt';

    // location of pages dir for stored content //
    $RAZOR['pages_dir'] = 'datastore/pages/';

    // location of media dir for stored content //
    $RAZOR['backup_dir'] = 'datastore/backup/';

    // location of blade packs //
    $RAZOR['bladepack_dir'] = 'blade_packs/';

    // name of homepage filename //
    $RAZOR['index_filename'] = 'index.php';

    // location of admin directory //
    $RAZOR['admin_dir'] = 'admin/';

    // name and location of admin filename //
    $RAZOR['admin_filename'] = 'admin/index.php';

    // name and location of admin config //
    $RAZOR['admin_config'] = 'admin/core/admin_config.php';

    // name and location of default public CSS file //
    $RAZOR['css_file_name'] = 'theme/default_css.css';

    // path to script directory to use in themes //
    $RAZOR['script_path'] = 'theme/scripts/';

    // default content name, sets the name of content loaded into the page //
    $RAZOR['default_content_name'] = 'slab';
    
    // default ext for filenames created for content //
    $RAZOR['default_file_ext'] = 'txt';

    // current version //
    $RAZOR['razor_current_ver'] = 'core v1.1 Stable';

    // sets amount of time max login attempts allowed in seconds eg 3600 = 60min //
    $RAZOR['razor_logat_time'] = 3600;

    // sets amount of failed login attempts allowed in time above //
    $RAZOR['razor_log_att'] = 8;

    // coning amount to reduce logs when checking, will not check more than this amount in the max time //
    $RAZOR['razor_log_amount'] = 60;
    
    // file manager path for sadmin //
    $RAZOR['razor_sadmin_path'] = '';
    
    // file manager path for admin and user //
    $RAZOR['razor_fileman_path'] = $RAZOR['datastore_dir'];

    // allowed file types to edit in file manager //
    $RAZOR['fileman_allowed_edit'] = 'txt,htm,html,php,htaccess,css';

    // allowed file types to view in file manager (document types only) //
    $RAZOR['fileman_allowed_view_doc'] = 'txt,pdf,htm,html,doc,xls,odt';

    // allowed file types to view in file manager (media types only) //
    $RAZOR['fileman_allowed_view_med'] = 'jpg,jpeg,png,gif,bmp';

    // allowed file types to view in file manager (media types only) //
    $RAZOR['html_charsets'] = 'ISO-8859-1,ISO-8859-2,ISO-2022-JP,EUC-KR,US-ASCII,UTF-8,SHIFT_JIS';

    ///////////////////////////////////////////////
    //  Assigning variables into data constants  //
    //          DO NOT EDIT THESE LINES          //
    ///////////////////////////////////////////////

    define( 'RAZOR_LOGS_DIR' , $RAZOR['logs_dir'] );
    define( 'RAZOR_FAILED_LOGIN_LOG' , $RAZOR['failed_logs'] );
    define( 'RAZOR_DATASTORE_DIR' , $RAZOR['datastore_dir'] );
    define( 'RAZOR_DATA' , $RAZOR['system_file'] );
    define( 'RAZOR_EXTENSIONS_ORDER' , "txt,php,htm,html" );
    define( 'RAZOR_PAGES_DIR' , $RAZOR['pages_dir'] );
    define( 'RAZOR_BACKUP_DIR' , $RAZOR['backup_dir'] );
    define( 'RAZOR_URL_FORMAT' , $RAZOR['index_filename'].'?'.$RAZOR['default_content_name'].'=%s' );
    define( 'RAZOR_ADMIN_DIR' , $RAZOR['admin_dir'] );
    define( 'RAZOR_ADMIN_FILENAME' , $RAZOR['admin_filename'] );
    define( 'RAZOR_ADMIN_CONFIG' , $RAZOR['admin_config'] );
    define( 'RAZOR_HOME_FILENAME' , $RAZOR['index_filename'] );
    define( 'RAZOR_BLADEPACK_DIR' , $RAZOR['bladepack_dir'] );
    define( 'RAZOR_CSS_FILE' , $RAZOR['css_file_name'] );
    define( 'RAZOR_SCRIPT_PATH' , $RAZOR['script_path'] );
    define( 'RAZOR_DEFAULT_CONTENT_NAME' , $RAZOR['default_content_name'] );
    define( 'RAZOR_DEFAULT_FILE_EXT' , $RAZOR['default_file_ext'] );
    define( 'RAZOR_CURRENT_VERSION' , $RAZOR['razor_current_ver'] );
    define( 'RAZOR_LOGAT_TIME' , $RAZOR['razor_logat_time'] );
    define( 'RAZOR_LOG_ATT' , $RAZOR['razor_log_att'] );
    define( 'RAZOR_LOG_AMOUNT' , $RAZOR['razor_log_amount'] );
    define( 'RAZOR_SADMIN_PATH' , $RAZOR['razor_sadmin_path'] );
    define( 'RAZOR_FILEMAN_PATH' , $RAZOR['razor_fileman_path'] );
    define( 'RAZOR_FILEMAN_EDIT_TYPE' , $RAZOR['fileman_allowed_edit'] );
    define( 'RAZOR_FILEMAN_VIEW_DOC' , $RAZOR['fileman_allowed_view_doc'] );
    define( 'RAZOR_FILEMAN_VIEW_MED' , $RAZOR['fileman_allowed_view_med'] );
    define( 'RAZOR_HTML_CHARSETS' , $RAZOR['html_charsets'] );
?>
